Skipped assertions in Jetty 12 tests

I noticed that tests for Jetty 12 adapter have Weld-related exclusions such as this one - https://github.com/arquillian/arquillian-container-jetty/blob/master/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedInContainerTestCase.java#L119-L121

This should be because you are using empty `beans.xml` which in CDI 4.0+ translates to discovery mode `annotated` and only beans with [bean defining annotations](https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0#bean_defining_annotations) will be discovered.

In other words, if you add `@Dependent` annotation onto your bean class ([here](https://github.com/arquillian/arquillian-container-jetty/blob/master/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/MyBean.java#L21)) it will be discovered. Alternatively, you could provide `beans.xml` which explicitly declares `all` discovery mode. Both approaches should get you to previous behavior.

That being said, doing so gave me a new stack that's not Weld related and that someone here (@joakime or @olamy) might know more about:

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.401 s <<< FAILURE! -- in org.jboss.arquillian.container.jetty.embedded_12_ee10.JettyEmbeddedInContainerTestCase
[ERROR] org.jboss.arquillian.container.jetty.embedded_12_ee10.JettyEmbeddedInContainerTestCase -- Time elapsed: 2.401 s <<< ERROR!
org.jboss.arquillian.container.spi.client.container.DeploymentException: Could not deploy 1a2f6627-44ff-4fd1-a990-56df29197acd.war
	at org.jboss.arquillian.container.jetty.embedded_12_ee10.JettyEmbeddedContainer.deploy(JettyEmbeddedContainer.java:273)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$3.call(ContainerDeployController.java:151)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$3.call(ContainerDeployController.java:118)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.executeOperation(ContainerDeployController.java:239)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.deploy(ContainerDeployController.java:118)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.container.impl.client.container.DeploymentExceptionHandler.verifyExpectedExceptionDuringDeploy(DeploymentExceptionHandler.java:47)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createDeploymentContext(ContainerDeploymentContextHandler.java:71)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createContainerContext(ContainerDeploymentContextHandler.java:54)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$1.perform(ContainerDeployController.java:92)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$1.perform(ContainerDeployController.java:77)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.forEachDeployment(ContainerDeployController.java:232)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.forEachManagedDeployment(ContainerDeployController.java:212)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.deployManaged(ContainerDeployController.java:77)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.execute(ContainerEventController.java:96)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.beforeClass(EventTestRunnerAdaptor.java:89)
	at org.jboss.arquillian.junit5.ArquillianExtension.beforeAll(ArquillianExtension.java:35)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.NullPointerException: Cannot invoke "jakarta.servlet.ServletContext.getContextPath()" because the return value of "org.eclipse.jetty.ee10.servlet.ServletHolder.getServletContext()" is null
	at org.jboss.arquillian.container.jetty.embedded_12_ee10.JettyEmbeddedContainer.deploy(JettyEmbeddedContainer.java:269)
	... 53 more
```

> ```
> Caused by: java.lang.NullPointerException: Cannot invoke "jakarta.servlet.ServletContext.getContextPath()" because the return value of "org.eclipse.jetty.ee10.servlet.ServletHolder.getServletContext()" is null
> 	at org.jboss.arquillian.container.jetty.embedded_12_ee10.JettyEmbeddedContainer.deploy(JettyEmbeddedContainer.java:269)
> 	... 53 more
> ```

That comes from using the wrong (not recommended) initialization of weld with Jetty 12.

Specifically the environment is using either a _Fallback_ or _DecoratingListener_ approach to initialize Weld.

There are many ways to initalize Weld with Jetty.

Take this example from Jetty 11.0.x

https://github.com/jetty/jetty-examples/blob/11.0.x/embedded/jersey-weld/src/main/java/examples/Main.java

The only one that should be used is the one labelled as `RECOMMENDED`.

If you look at that same example, but for Jetty 12, you'll see that options for `FALLBACK` and `DECORATING_LISTENER` have been removed.

https://github.com/jetty/jetty-examples/blob/12.0.x/embedded/ee10-jersey-weld/src/main/java/examples/Main.java

--- a/jetty-embedded-12-ee10/pom.xml
+++ b/jetty-embedded-12-ee10/pom.xml
@@ -15,8 +15,7 @@
   <description>Jetty Embedded 12.0.x container integration for the Arquillian project</description>
 
   <properties>
-    <!-- when upgrading check if this fix JettyEmbeddedInContainerTestCase#shouldBeAbleToInjectMembersIntoTestClass -->
-    <version.weld>5.1.0.Final</version.weld>
+    <version.weld>5.1.2.Final</version.weld>
     <jakarta.servlet.api.version>6.0.0</jakarta.servlet.api.version>
 
     <!-- Jetty 12 need Java 17 -->
@@ -147,6 +146,11 @@
       <artifactId>alpn-api</artifactId>
       <version>1.1.3.v20160715</version>
     </dependency>
+    <!-- Needed for Jetty be able to perform injection into servlets/filters etc -->
+    <dependency>
+      <groupId>org.eclipse.jetty.ee10</groupId>
+      <artifactId>jetty-ee10-cdi</artifactId>
+    </dependency>
 
     <dependency>
       <groupId>org.jboss.arquillian.junit5</groupId>
@@ -237,7 +241,6 @@
     <dependency>
       <groupId>org.jboss.weld.servlet</groupId>
       <artifactId>weld-servlet-core</artifactId>
-      <scope>test</scope>
     </dependency>
 
     <dependency>

--- a/jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedContainer.java
+++ b/jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedContainer.java
@@ -29,6 +29,8 @@ import org.codehaus.plexus.util.ReflectionUtils;
 import org.eclipse.jetty.deploy.App;
 import org.eclipse.jetty.deploy.AppLifeCycle;
 import org.eclipse.jetty.deploy.DeploymentManager;
+import org.eclipse.jetty.ee10.cdi.CdiDecoratingListener;
+import org.eclipse.jetty.ee10.cdi.CdiServletContainerInitializer;
 import org.eclipse.jetty.http.CookieCompliance;
 import org.eclipse.jetty.http.HttpVersion;
 import org.eclipse.jetty.http2.server.HTTP2CServerConnectionFactory;
@@ -252,6 +254,15 @@ public class JettyEmbeddedContainer implements DeployableContainer<JettyEmbedded
             App app = appProvider.createApp(archive);
             deployer.removeApp(app);
             WebAppContext webAppContext = getWebAppContext(app);
+
+            // Jetty setup telling Jetty's CdiDecoratingListener how to operate.
+            webAppContext.setInitParameter(CdiServletContainerInitializer.CDI_INTEGRATION_ATTRIBUTE, CdiDecoratingListener.MODE);
+            // jetty setup for layer between Weld and Jetty.
+            webAppContext.addServletContainerInitializer(new CdiServletContainerInitializer());
+            // Weld's org.jboss.weld.environment.servlet.EnhancedListener can be discovered automatically
+            // However, it won't happen if jetty-ee10-annotations JAR isn't present in runtime, hence we add it explicitly
+            // The listener will start up Weld container so long as there is an archive with any beans in it
+            webAppContext.addServletContainerInitializer(new org.jboss.weld.environment.servlet.EnhancedListener());
 
             if (containerConfig.areMimeTypesSet()) {
                 containerConfig.getMimeTypes().forEach((s, s2) -> webAppContext.getMimeTypes().addMimeMapping(s, s2));

--- a/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedClientTestCase.java
+++ b/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedClientTestCase.java
@@ -30,11 +30,14 @@ import org.jboss.arquillian.container.test.api.OperateOnDeployment;
 import org.jboss.arquillian.container.test.api.TargetsContainer;
 import org.jboss.arquillian.junit5.ArquillianExtension;
 import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.GenericArchive;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
 import org.jboss.shrinkwrap.api.asset.StringAsset;
 import org.jboss.shrinkwrap.api.spec.WebArchive;
 import org.jboss.shrinkwrap.descriptor.api.Descriptors;
 import org.jboss.shrinkwrap.descriptor.api.webapp30.WebAppDescriptor;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
@@ -59,15 +62,17 @@ public class JettyEmbeddedClientTestCase {
     @Deployment(testable = false)
     public static WebArchive getTestArchive() {
         return ShrinkWrap.create(WebArchive.class, "client-http.war")
-            .addClass(MyServlet.class)
+            .addClass(MyOtherServlet.class)
+            .addClass(MyOtherBean.class)
+            .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
             .setWebXML(new StringAsset(Descriptors.create(WebAppDescriptor.class)
                 .version("4.0")
                 .createServlet()
-                .servletClass(MyServlet.class.getName())
-                .servletName("MyServlet").up()
+                .servletClass(MyOtherServlet.class.getName())
+                .servletName("MyOtherServlet").up()
                 .createServletMapping()
-                .servletName("MyServlet")
-                .urlPattern(MyServlet.URL_PATTERN).up()
+                .servletName("MyOtherServlet")
+                .urlPattern(MyOtherServlet.URL_PATTERN).up()
                 .exportAsString()));
     }
 
@@ -119,12 +124,12 @@ public class JettyEmbeddedClientTestCase {
     @Test
     public void shouldBeAbleToInvokeServletInDeployedWebApp() throws Exception {
 
-        String body = httpClient.GET(new URL(url, MyServlet.URL_PATTERN).toURI()).getContentAsString();
+        String body = httpClient.GET(new URL(url, MyOtherServlet.URL_PATTERN).toURI()).getContentAsString();
 
         assertThat(
             "Verify that the servlet was deployed and returns expected result",
             body,
-            Matchers.is(MyServlet.MESSAGE));
+            Matchers.is(MyOtherServlet.MESSAGE));
     }
 
     @Test

--- a/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedInContainerTestCase.java
+++ b/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedInContainerTestCase.java
@@ -115,9 +115,8 @@ public class JettyEmbeddedInContainerTestCase {
         try (Connection c = ds.getConnection()) {
             assertThat(c.getMetaData().getDatabaseProductName(), is("H2"));
         }
-//      FIXME not working anymore with weld 5.1.0.Final
-//        assertThat(testBean, notNullValue());
-//        assertThat(testBean.getName(), is("Jetty"));
+        assertThat(testBean, notNullValue());
+        assertThat(testBean.getName(), is("Jetty"));
     }
 
     @Test

--- a/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/MyBean.java
+++ b/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/MyBean.java
@@ -17,7 +17,9 @@
 package org.jboss.arquillian.container.jetty.embedded_12_ee10;
 
 import jakarta.annotation.Resource;
+import jakarta.enterprise.context.Dependent;
 
+@Dependent
 public class MyBean {
     @Resource(name = "name")
     private String name;

--- a/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/MyOtherBean.java
+++ b/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/MyOtherBean.java
@@ -1,0 +1,11 @@
+package org.jboss.arquillian.container.jetty.embedded_12_ee10;
+
+import jakarta.enterprise.context.Dependent;
+
+@Dependent
+public class MyOtherBean {
+
+    public String ping() {
+        return MyOtherBean.class.getSimpleName();
+    }
+}

--- a/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/MyOtherServlet.java
+++ b/jetty-embedded-12-ee10/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/MyOtherServlet.java
@@ -1,0 +1,30 @@
+package org.jboss.arquillian.container.jetty.embedded_12_ee10;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import jakarta.inject.Inject;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+public class MyOtherServlet extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+
+    public static final String URL_PATTERN = "Test2";
+
+    public static final String MESSAGE = "hey there";
+
+    @Inject
+    MyOtherBean bean;
+
+    @Override
+    protected void service(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        assertThat(bean.ping(), is(MyOtherBean.class.getSimpleName()));
+        response.getWriter().append(MESSAGE);
+    }
+}

--- a/jetty-embedded-12-ee10/src/test/resources/in-container-web.xml
+++ b/jetty-embedded-12-ee10/src/test/resources/in-container-web.xml
@@ -17,8 +17,4 @@
     <env-entry-value>Remote</env-entry-value>
   </env-entry>
 
-  <listener>
-    <listener-class>org.jboss.weld.environment.servlet.Listener</listener-class>
-  </listener>
-
 </web-app>

--- a/jetty-embedded-12-ee9/pom.xml
+++ b/jetty-embedded-12-ee9/pom.xml
@@ -15,9 +15,7 @@
   <description>Jetty Embedded 12.0.x container integration for the Arquillian project</description>
 
   <properties>
-    <!-- when upgrading check if this fix JettyEmbeddedInContainerTestCase#shouldBeAbleToInjectMembersIntoTestClass -->
-    <version.weld>5.1.0.Final</version.weld>
-<!--    <jakarta.servlet.api.version>6.0.0</jakarta.servlet.api.version>-->
+    <version.weld>5.1.2.Final</version.weld>
 
     <!-- Jetty 12 need Java 17 -->
     <maven.compiler.target>17</maven.compiler.target>
@@ -31,7 +29,6 @@
         <groupId>org.jboss.weld.servlet</groupId>
         <artifactId>weld-servlet-core</artifactId>
         <version>${version.weld}</version>
-        <scope>test</scope>
       </dependency>
       <dependency>
         <groupId>org.eclipse.jetty</groupId>
@@ -143,6 +140,11 @@
       <artifactId>alpn-api</artifactId>
       <version>1.1.3.v20160715</version>
     </dependency>
+    <!-- Needed for Jetty be able to perform injection into servlets/filters etc -->
+    <dependency>
+      <groupId>org.eclipse.jetty.ee9</groupId>
+      <artifactId>jetty-ee9-cdi</artifactId>
+    </dependency>
 
     <dependency>
       <groupId>org.jboss.arquillian.junit5</groupId>
@@ -225,7 +227,6 @@
     <dependency>
       <groupId>org.jboss.weld.servlet</groupId>
       <artifactId>weld-servlet-core</artifactId>
-      <scope>test</scope>
     </dependency>
 
     <dependency>

--- a/jetty-embedded-12-ee9/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/JettyEmbeddedContainer.java
+++ b/jetty-embedded-12-ee9/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/JettyEmbeddedContainer.java
@@ -29,6 +29,8 @@ import org.codehaus.plexus.util.ReflectionUtils;
 import org.eclipse.jetty.deploy.App;
 import org.eclipse.jetty.deploy.AppLifeCycle;
 import org.eclipse.jetty.deploy.DeploymentManager;
+import org.eclipse.jetty.ee9.cdi.CdiDecoratingListener;
+import org.eclipse.jetty.ee9.cdi.CdiServletContainerInitializer;
 import org.eclipse.jetty.http.CookieCompliance;
 import org.eclipse.jetty.http.HttpVersion;
 import org.eclipse.jetty.http2.server.HTTP2CServerConnectionFactory;
@@ -252,6 +254,15 @@ public class JettyEmbeddedContainer implements DeployableContainer<JettyEmbedded
             deployer.removeApp(app);
              appProvider.createContextHandler(app);
             WebAppContext webAppContext = getWebAppContext(app);
+
+            // Jetty setup telling Jetty's CdiDecoratingListener how to operate.
+            webAppContext.setInitParameter(CdiServletContainerInitializer.CDI_INTEGRATION_ATTRIBUTE, CdiDecoratingListener.MODE);
+            // jetty setup for layer between Weld and Jetty.
+            webAppContext.addServletContainerInitializer(new CdiServletContainerInitializer());
+            // Weld's org.jboss.weld.environment.servlet.EnhancedListener can be discovered automatically
+            // However, it won't happen if jetty-ee9-annotations JAR isn't present in runtime, hence we add it explicitly
+            // The listener will start up Weld container so long as there is an archive with any beans in it
+            webAppContext.addServletContainerInitializer(new org.jboss.weld.environment.servlet.EnhancedListener());
 
             if (containerConfig.areMimeTypesSet()) {
                 containerConfig.getMimeTypes().forEach((s, s2) -> webAppContext.getMimeTypes().addMimeMapping(s, s2));

--- a/jetty-embedded-12-ee9/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/JettyEmbeddedClientTestCase.java
+++ b/jetty-embedded-12-ee9/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/JettyEmbeddedClientTestCase.java
@@ -31,6 +31,7 @@ import org.jboss.arquillian.container.test.api.TargetsContainer;
 import org.jboss.arquillian.junit5.ArquillianExtension;
 import org.jboss.arquillian.test.api.ArquillianResource;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
 import org.jboss.shrinkwrap.api.asset.StringAsset;
 import org.jboss.shrinkwrap.api.spec.WebArchive;
 import org.jboss.shrinkwrap.descriptor.api.Descriptors;
@@ -59,15 +60,17 @@ public class JettyEmbeddedClientTestCase {
     @Deployment(testable = false)
     public static WebArchive getTestArchive() {
         return ShrinkWrap.create(WebArchive.class, "client-http.war")
-            .addClass(MyServlet.class)
+            .addClass(MyOtherServlet.class)
+            .addClass(MyOtherBean.class)
+            .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
             .setWebXML(new StringAsset(Descriptors.create(WebAppDescriptor.class)
                 .version("4.0")
                 .createServlet()
-                .servletClass(MyServlet.class.getName())
-                .servletName("MyServlet").up()
+                .servletClass(MyOtherServlet.class.getName())
+                .servletName("MyOtherServlet").up()
                 .createServletMapping()
-                .servletName("MyServlet")
-                .urlPattern(MyServlet.URL_PATTERN).up()
+                .servletName("MyOtherServlet")
+                .urlPattern(MyOtherServlet.URL_PATTERN).up()
                 .exportAsString()));
     }
 
@@ -119,12 +122,12 @@ public class JettyEmbeddedClientTestCase {
     @Test
     public void shouldBeAbleToInvokeServletInDeployedWebApp() throws Exception {
 
-        String body = httpClient.GET(new URL(url, MyServlet.URL_PATTERN).toURI()).getContentAsString();
+        String body = httpClient.GET(new URL(url, MyOtherServlet.URL_PATTERN).toURI()).getContentAsString();
 
         assertThat(
             "Verify that the servlet was deployed and returns expected result",
             body,
-            Matchers.is(MyServlet.MESSAGE));
+            Matchers.is(MyOtherServlet.MESSAGE));
     }
 
     @Test

--- a/jetty-embedded-12-ee9/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/JettyEmbeddedInContainerTestCase.java
+++ b/jetty-embedded-12-ee9/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/JettyEmbeddedInContainerTestCase.java
@@ -116,9 +116,8 @@ public class JettyEmbeddedInContainerTestCase {
         try (Connection c = ds.getConnection()) {
             assertThat(c.getMetaData().getDatabaseProductName(), is("H2"));
         }
-//      FIXME not working anymore with weld 5.1.0.Final
-//        assertThat(testBean, notNullValue());
-//        assertThat(testBean.getName(), is("Jetty"));
+        assertThat(testBean, notNullValue());
+        assertThat(testBean.getName(), is("Jetty"));
     }
 
     @Test

--- a/jetty-embedded-12-ee9/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/MyBean.java
+++ b/jetty-embedded-12-ee9/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/MyBean.java
@@ -17,7 +17,9 @@
 package org.jboss.arquillian.container.jetty.embedded_12_ee9;
 
 import jakarta.annotation.Resource;
+import jakarta.enterprise.context.Dependent;
 
+@Dependent
 public class MyBean {
     @Resource(name = "name")
     private String name;

--- a/jetty-embedded-12-ee9/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/MyOtherBean.java
+++ b/jetty-embedded-12-ee9/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/MyOtherBean.java
@@ -1,0 +1,11 @@
+package org.jboss.arquillian.container.jetty.embedded_12_ee9;
+
+import jakarta.enterprise.context.Dependent;
+
+@Dependent
+public class MyOtherBean {
+
+    public String ping() {
+        return MyOtherBean.class.getSimpleName();
+    }
+}

--- a/jetty-embedded-12-ee9/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/MyOtherServlet.java
+++ b/jetty-embedded-12-ee9/src/test/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/MyOtherServlet.java
@@ -1,0 +1,30 @@
+package org.jboss.arquillian.container.jetty.embedded_12_ee9;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import jakarta.inject.Inject;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+public class MyOtherServlet extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+
+    public static final String URL_PATTERN = "Test2";
+
+    public static final String MESSAGE = "hey there";
+
+    @Inject
+    MyOtherBean bean;
+
+    @Override
+    protected void service(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        assertThat(bean.ping(), is(MyOtherBean.class.getSimpleName()));
+        response.getWriter().append(MESSAGE);
+    }
+}

--- a/jetty-embedded-12-ee9/src/test/resources/in-container-web.xml
+++ b/jetty-embedded-12-ee9/src/test/resources/in-container-web.xml
@@ -17,8 +17,4 @@
     <env-entry-value>Remote</env-entry-value>
   </env-entry>
 
-  <listener>
-    <listener-class>org.jboss.weld.environment.servlet.Listener</listener-class>
-  </listener>
-
 </web-app>

